RNASeqPipeline.Rmd

```diff
@@ -6,139 +6,129 @@ output: pdf_document
 
 ```{r setup, include=FALSE}
 knitr::opts_chunk$set(echo = TRUE)
-```
-
-```{r}
-install.packages("BiocManager")
-BiocManager::install("DESeq2")
-```
 
-```{r}
+# Load libraries
 library(DESeq2)
 library(onlineFDR)
 library(seqgendiff)
 library(compcodeR)
+```
 
-generateSequentialCountMatrix <- function(numHyp, varNum, diffExp){
+```{r simulate_data_fct}
+generateSequentialCountMatrix <- function(numHyp, varNum, diffExp, nSamples = 5){
   
-  runningMatrix <- matrix()
-  condList <- c()
-  diffExpression <- c()
+  # numHyp: ?
+  # varNum: ?
+  # diffExp: ?
   
-  for (i in 1:numHyp){
+  # initialize a list of datasets over time
+  synthData <- list()
   
-  synthData <- generateSyntheticData(dataset = "B_625_625", n.vars = varNum, 
-                                   samples.per.cond = 5, n.diffexp = diffExp, 
-                                   repl.id = 1, seqdepth = 1e7, 
-                                   fraction.upregulated = 0.5, 
-                                   between.group.diffdisp = FALSE, 
-                                   filter.threshold.total = -1, 
-                                   filter.threshold.mediancpm = -1, 
-                                   fraction.non.overdispersed = 0, 
-                                   output.file = "OutputTestSynth.rds")
+  #describe your conditions
+  condVec <- c(rep("control", nSamples), rep("treatment", nSamples)) 
   
- diffExpression <- c(diffExpression, synthData@variable.annotations$differential.expression)
-  
-  condList <- c(condList, "C","C", "C", "C","C","T","T","T","T","T") 
-  if (i == 1){
-    runningMatrix <- synthData@count.matrix
-  }
-  else{
-     runningMatrix <- cbind(runningMatrix, synthData@count.matrix)
-     
-     
-      
+  for (i in seq_along(1:numHyp)){
+    
+    # you need a seed for reproducibility of simulated data
+    set.seed(i)
+    
+    synthData[[i]] <- generateSyntheticData(dataset = "B_625_625", 
+                                            n.vars = varNum, 
+                                            samples.per.cond = nSamples, 
+                                            n.diffexp = diffExp, 
+                                            repl.id = 1, 
+                                            seqdepth = 1e7, 
+                                            fraction.upregulated = 0.5, 
+                                            between.group.diffdisp = FALSE, 
+                                            filter.threshold.total = -1, # don't understand this
+                                            filter.threshold.mediancpm = -1, # don't understand this
+                                            fraction.non.overdispersed = 0, 
+                                            output.file = "OutputTestSynth.rds")
```

Comment:
I'd also rethink this. At the moment, you'd be overwriting your synthetic dataset at each iteration, which is likely not what you intended. If you wanted to save each dataset, you could do paste0("OutputtestSynth", i, ".rds"), but I'd also encourage you to think if this is what you want to end up saving.